src/server/routes/api/document/index.js

```diff
@@ -147,6 +214,37 @@ const checkApiKey = (apiKey) => {
   }
 };
 
+// create the demo doc
+http.post('/demo', function(req, res, next){
+  let meta = _.assign({}, req.body, { id: DEMO_ID });
+  let secret = DEMO_SECRET;
+
+  ( tryPromise(() => createSecret({ secret }))
+    .then(loadTables)
+    .then(({ docDb, eleDb }) => createDoc({ docDb, eleDb, secret, meta }))
+    .then(getDocJson)
+    .then(json => res.json(json))
+    .catch(err => next(err))
+  );
+});
+
+// delete the demo doc
+http.delete('/demo', function(req, res, next){
+  let { apiKey } = req.body;
+  let secret = DEMO_SECRET;
+
+  let clearDemoRows = table => table.filter({ secret }).delete();
+
+  ( tryPromise(() => checkApiKey(apiKey))
+    .then(loadTables)
+    .then(({ docTable, eleTable }) => (
+      Promise.all([docTable, eleTable].map(clearDemoRows))
+    ))
```

Comment:
Its complaining that `TypeError: Cannot read property 'filter' of undefined` for `factoid/src/server/routes/api/document/index.js:236:38`